backend/src/apiserver/server/util.go

```diff
@@ -29,17 +29,24 @@ import (
 )
 
 func loadFile(fileReader io.Reader, MaxFileLength int) ([]byte, error) {
-	reader := bufio.NewReader(fileReader)
-	pipelineFile := make([]byte, MaxFileLength+1)
-	size, err := reader.Read(pipelineFile)
-	if err != nil && err != io.EOF {
-		return nil, util.NewInvalidInputErrorWithDetails(err, "Error read pipeline file")
+	reader := bufio.NewReaderSize(fileReader, MaxFileLength)
+	var pipelineFile []byte
+	for {
```

Comment:
Let's add a TODO comment to improve the performance by setting a custom buffer size instead of 64KB. With `MaxFileLength` of 32MB we may end up with 500 iterations in this loop instead of reading the file in a single step.